### PR TITLE
[CARBONDATA-1238] Decouple the datatype convert from Spark code in core module

### PR DESCRIPTION
**Pull request by chenliang613**
Decouple the datatype convert from Spark code in core module：
1.Use decimal(org.apache.spark.sql.types.Decimal.apply()) in Spark engine, use java's decimal in other engines.
2.Use org.apache.spark.unsafe.types.UTF8String in Spark engine, use String in other engines.

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3193/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/598/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3194/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/599/

File: hadoop/src/main/java/org/apache/carbondata/hadoop/CarbonInputFormat.java [hunk 1]
```diff
@@ -768,29 +818,6 @@ public QueryModel getQueryModel(InputSplit inputSplit, TaskAttemptContext taskAt
     return queryModel;
   }
 
-  public CarbonReadSupport<T> getReadSupportClass(Configuration configuration) {
```
Comment (jackylk) [inline hadoop/src/main/java/org/apache/carbondata/hadoop/CarbonInputFormat.java]: any change for this method?

File: integration/spark-common/src/main/java/org/apache/carbondata/spark/util/SparkDataTypeConverterImp.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import java.io.Serializable;
+
+import org.apache.carbondata.core.util.DataTypeConverter;
+
+import org.apache.spark.unsafe.types.UTF8String;
+
+/**
+ * Convert java data type to spark data type
+ */
+public final class SparkDataTypeConverterImp implements DataTypeConverter, Serializable {
```
Comment (jackylk) [inline integration/spark-common/src/main/java/org/apache/carbondata/spark/util/SparkDataTypeConverterImp.java]: type, `SparkDataTypeConverterImpl`

File: hadoop/src/main/java/org/apache/carbondata/hadoop/CarbonInputFormat.java (hunk 1 above)
Comment (chenliang613) [inline hadoop/src/main/java/org/apache/carbondata/hadoop/CarbonInputFormat.java]: no any code change, just move "set and get" method to together.  for example : put setFilterPredicates and getFilterPredicates to together.

File: integration/spark-common/src/main/java/org/apache/carbondata/spark/util/SparkDataTypeConverterImp.java (hunk 2 above)
Comment (chenliang613) [inline integration/spark-common/src/main/java/org/apache/carbondata/spark/util/SparkDataTypeConverterImp.java]: ok

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/613/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3208/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/616/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3211/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/631/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3227/

**Comment by chenliang613** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3232/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/637/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3248/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/653/

**Comment by jackylk** [comment]
LGTM

**Pull request closed without merging**